pkg/platform/kube/provisioner/apigateway/provisioner.go

```diff
@@ -0,0 +1,369 @@
+package apigateway
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/platform"
+	nuclioio "github.com/nuclio/nuclio/pkg/platform/kube/apis/nuclio.io/v1beta1"
+	nuclioio_client "github.com/nuclio/nuclio/pkg/platform/kube/client/clientset/versioned"
+	"github.com/nuclio/nuclio/pkg/platform/kube/ingress"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Provisioner struct {
+	Logger          logger.Logger
+	kubeClientSet   kubernetes.Interface
+	nuclioClientSet nuclioio_client.Interface
+	ingressManager  *ingress.Manager
+}
+
+func NewProvisioner(loggerInstance logger.Logger,
+	kubeClientSet kubernetes.Interface,
+	nuclioClientSet nuclioio_client.Interface,
+	ingressManager *ingress.Manager) (*Provisioner, error) {
+
+	newProvisioner := &Provisioner{
+		Logger:          loggerInstance.GetChild("apigateway"),
+		kubeClientSet:   kubeClientSet,
+		nuclioClientSet: nuclioClientSet,
+		ingressManager:  ingressManager,
+	}
+
+	return newProvisioner, nil
+}
+
+func (p *Provisioner) CreateOrUpdateAPIGateway(ctx context.Context, apiGateway *nuclioio.NuclioAPIGateway) error {
```

Comment:
Again, this is under `apigateway`, unless other resources is being create/update, I believe the `APIGateway` is redundant